spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -36,36 +37,44 @@
  */
 public class ChannelBindingService {
 
-	private final Binder<MessageChannel> binder;
+	private BinderFactory<MessageChannel> binderFactory;
 
 	private final ChannelBindingServiceProperties channelBindingServiceProperties;
 
-	public ChannelBindingService(ChannelBindingServiceProperties channelBindingServiceProperties, Binder<MessageChannel> binder) {
+	public ChannelBindingService(ChannelBindingServiceProperties channelBindingServiceProperties,
+								 BinderFactory<MessageChannel> binderFactory) {
 		this.channelBindingServiceProperties = channelBindingServiceProperties;
-		this.binder = binder;
+		this.binderFactory = binderFactory;
 	}
 
 	public void bindConsumer(MessageChannel inputChannel, String inputChannelName) {
 		String channelBindingTarget = this.channelBindingServiceProperties.getBindingDestination(inputChannelName);
+		String transport = this.channelBindingServiceProperties.getBinder(inputChannelName);
+		Binder<MessageChannel> binder = binderFactory.getBinder(transport);
 		if (BinderUtils.isChannelPubSub(channelBindingTarget)) {
-			this.binder.bindPubSubConsumer(removePrefix(channelBindingTarget),
+			BindingProperties bindingProperties = this.channelBindingServiceProperties.getBindings()
+					.get(inputChannelName);
+			String group = bindingProperties == null ? null : bindingProperties.getGroup();
```

Comment:
Right - that code was moved to `consumerGroup(inputChannelName)` (below) with the fix for #211